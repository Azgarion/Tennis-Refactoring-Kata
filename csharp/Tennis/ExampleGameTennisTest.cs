using ApprovalTests.Reporters;
using NUnit.Framework;
using Tennis.Domain;

namespace Tennis.Tests
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ExampleGameTennisTest 
    {
        [Test]
        public void CheckGame1()
        {
            using (new GoldenMasterTest())
            {
                var game = new TennisGame1("player1", "player2");
                RealisticTennisGame(game);
            }
        }

        [Test]
        public void CheckGame2()
        {
            using (new GoldenMasterTest())
            {
                var game = new TennisGame2("player1", "player2");
                RealisticTennisGame(game);
            }
        }

        [Test]
        public void CheckGame3()
        {
            using (new GoldenMasterTest())
            {
                var game = new TennisGame3("player1", "player2");
                RealisticTennisGame(game);
            }
        }

        private void RealisticTennisGame(ITennisGame game)
        {
            string[] points = { "player1", "player1", "player2", "player2", "player1", "player1" };
            string[] expectedScores = { "Fifteen-Love", "Thirty-Love", "Thirty-Fifteen", "Thirty-All", "Forty-Thirty", "Win for player1" };
            for (var i = 0; i < 6; i++)
            {
                game.WonPoint(points[i]);
                Assert.AreEqual(expectedScores[i], game.GetScore());
            }
        }
    }
}
