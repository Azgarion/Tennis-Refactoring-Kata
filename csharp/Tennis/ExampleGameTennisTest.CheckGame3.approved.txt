New TennisGame3
Player1Name equals player1
Player2Name equals player2
Player player1 score incremented to 1
GetScore returned Fifteen-Love
Player player1 score incremented to 2
GetScore returned Thirty-Love
Player player2 score incremented to 1
GetScore returned Thirty-Fifteen
Player player2 score incremented to 2
GetScore returned Thirty-All
Player player1 score incremented to 3
GetScore returned Forty-Thirty
Player player1 score incremented to 4
GetScore returned Win for player1
